### DISCUSSION TITLE
possible to get command line args

### DISCUSSION THREAD

**Discussion by kenyee** [quarkusio/quarkus#23819] [Q&A] [answered]
Our deployment environment launches jars using command line args instead of system env vars like this:

```
-jar myapp.jar -log.level=DEBUG -https.port=:31760 -http.port=:31048 -admin.port=:31126 -service.identifier=myapp:devel:cluster1 
```
Is it possible to get these in Quarkus?

I thought subclassing javax.ws.rs.core.Application might give me access to the command line arguments (no luck...can only access a list of classes, properties, and singletons) but ideally, I'd want to get these in these as part of quarkus config resources: https://quarkus.io/guides/config-reference#configuration-sources

**Comment by kenyee** [comment]
nvm...found a workaround for this to put the ports in system properties

**Comment by radcortez** [comment, marked as answer]
Did you try with Quarkus Main method: https://quarkus.io/guides/lifecycle#the-main-method? You should be able to collect the args into some static structure and then expose it via a `ConfigSource`.

**Reply by kenyee** [reply]
nice...didn't know about that.  It would have worked :-)
